gcc-11.3.0/libcpp/mkdeps.c

```diff
@@ -725,16 +726,255 @@ calculate_sha256_omnibor_with_contents (std::string contents,
   delete [] init_data_char_array;
 }
 
+/* OmniBOR structure which represents a pair of a dependency filename and
+   its gitoid.  */
+
+class omnibor_dep
+{
+public:
+
+  omnibor_dep (std::string name1, std::string gitoid1)
+    : name (name1), gitoid (gitoid1)
+  {}
+  ~omnibor_dep ()
+  {}
+
+  std::string name;
+  std::string gitoid;
+};
+
+static bool
+omnibor_cmp (const class omnibor_dep *first, const class omnibor_dep *second)
+{
+  return first->gitoid < second->gitoid;
+}
+
+/* Get the path of the directory where the resulting output file will be
+   stored.  Also, fill the memory pointed to by is_E_or_S_or_c with 3 if
+   -E option is specified, 2 if -S option is specified, 1 if -c option
+   is specified, or 0 otherwise (if the linking is done as well).  */
+
```

Comment:
Instead of having numbers 3, 2, 1 and 0 representing the type of process GCC is doing, create an enum and use its values.